crates/swc/tests/tsc-references/controlFlowAssignmentPatternOrder.2.minified.js

```diff
@@ -37,19 +37,19 @@
 }// same as above but on left of a binary expression
 {
     let b, a = 0;
-    [{ 1: b } = [
+    [{ [a = 1]: b } = [
```

Comment:
Oh this is a bugfix

Reply:
Yeah. This PR arises from the problem that assign happens after the evaluation of LHS and RHS, so it's impossible to report reference and assign at the same time.